src/lib.rs

```diff
@@ -176,7 +173,7 @@ pub struct SystemContextInner<TYPES: NodeType, I: NodeImplementation<TYPES>> {
 #[derive(Clone)]
 pub struct SystemContext<CONSENSUS: ConsensusType, TYPES: NodeType, I: NodeImplementation<TYPES>> {
     /// Handle to internal hotshot implementation
-    inner: Arc<SystemContextInner<TYPES, I>>,
+    pub inner: Arc<SystemContextInner<TYPES, I>>,
```

Comment:
hmm what's the reason for this? If we need this long term, maybe `pub crate` or a getter would suffice?

Reply:
It's a hack to allow for the unit test stuff

Reply:
So yes we could add getters instead